src/Futhark/Fmt/Monad.hs

```diff
@@ -314,143 +311,140 @@ hardline = do
       pure P.line
 
 -- | A line or a space depending on layout.
-line :: FmtM Fmt
+line :: Fmt
 line = space <|> hardline
 
 -- | A comment.
-comment :: T.Text -> FmtM Fmt
+comment :: T.Text -> Fmt
 comment c = do
   modify (\s -> s {lastOutput = Just Line})
   pure $ P.pretty c <> P.line
 
 -- in order to handle trailing comments its VERY important to
 -- evaluate the seperator after each element in the list.
-sep :: (Format a, Format b) => a -> [b] -> FmtM Fmt
+sep :: Fmt -> [Fmt] -> Fmt
 sep _ [] = nil
-sep s (a : as) = auxiliary (fmt a) as
+sep s (a : as) = auxiliary a as
   where
     auxiliary acc [] = acc
     auxiliary acc (x : xs) =
       auxiliary (acc <:> prependSepComments s (prependSepComments x (s <:> x))) xs
 
 -- | Seperates element by a @s@ followed by a space in singleline layout and
 -- seperates by a line followed by a @s@ in multine layout.
-sepLine :: (Format a, Format b) => a -> [b] -> FmtM Fmt
+sepLine :: Fmt -> [Fmt] -> Fmt
 sepLine s = sep (s <:> space <|> hardline <:> s)
 
 -- | This is used for function arguments. It seperates multiline arguments by
 -- lines and singleline arguments by spaces.
-sepArgs :: (Format a) => [a] -> FmtM Fmt
+sepArgs :: [Fmt] -> Fmt
```

Comment:
sepArgs needs to know the Location of each argument in order to format this probably.